src/vsphere_cpi/lib/cloud/vsphere/cloud.rb

```diff
@@ -276,11 +276,21 @@ def create_vm(agent_id, stemcell_cid, vm_type, networks_spec, existing_disk_cids
 
         begin
           if @config.nsxt_enabled?
-            @nsxt_provider.add_vm_to_nsgroups(created_vm, vm_type.nsxt)
+            ns_groups = vm_type.ns_groups || []
```

Comment:
This logic is complicated enough that I'd rather move this stuff into a separate module and have a stack of modules like middleware.